exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/_mapping.py

```diff
@@ -0,0 +1,135 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from re import UNICODE, compile
+
+_SANITIZE_NAME_RE = compile(r"([^a-zA-Z0-9:]+)|_{2,}", UNICODE)
+# Same as name, but doesn't allow ":"
+_SANITIZE_ATTRIBUTE_KEY_RE = compile(r"([^a-zA-Z0-9]+)|_{2,}", UNICODE)
+
+# UCUM annotations are ASCII chars 33-126 enclosed in curly braces
+# https://ucum.org/ucum#para-6
+_UCUM_ANNOTATION_CURLY = compile(r"{[!-~]*}")
+
+# Remaps common UCUM and SI units to prometheus conventions. Copied from
+# https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.101.0/pkg/translator/prometheus/normalize_name.go#L19
+# See specification:
+# https://github.com/open-telemetry/opentelemetry-specification/blob/v1.33.0/specification/compatibility/prometheus_and_openmetrics.md#metric-metadata-1
+_UNIT_MAPPINGS = {
+    # Time
+    "d": "days",
+    "h": "hours",
+    "min": "minutes",
+    "s": "seconds",
+    "ms": "milliseconds",
+    "us": "microseconds",
+    "ns": "nanoseconds",
+    # Bytes
+    "By": "bytes",
+    "KiBy": "kibibytes",
+    "MiBy": "mebibytes",
+    "GiBy": "gibibytes",
+    "TiBy": "tibibytes",
+    "KBy": "kilobytes",
+    "MBy": "megabytes",
+    "GBy": "gigabytes",
+    "TBy": "terabytes",
+    # SI
+    "m": "meters",
+    "V": "volts",
+    "A": "amperes",
+    "J": "joules",
+    "W": "watts",
+    "g": "grams",
+    # Misc
+    "Cel": "celsius",
+    "Hz": "hertz",
+    # TODO: this conflicts with the spec but I think it is correct. Need to open a spec issue
+    "1": "",
+    "%": "percent",
+}
+# Similar to _UNIT_MAPPINGS, but for "per" unit denominator.
+# Example: s => per second (singular)
+# Copied from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/80317ce83ed87a2dff0c316bb939afbfaa823d5e/pkg/translator/prometheus/normalize_name.go#L58
+_PER_UNIT_MAPPINGS = {
```

Comment:
Why only time units are here?

Reply:
In the below logic, if the unit isn't found in mapping, it is used instead.

> _PER_UNIT_MAPPINGS.get(bottom, bottom)

Reply:
> It's also weird that the m maps to minute when above it maps to meters.

That seems incorrect. @bertysentry do you remember why we did this?

Reply:
Yes all this logic looks a bit surprising, but it's just a practical conversion that covers all real use cases, notably the units defined in receivers (at the time it was written).

`By/m` was more likely to mean `bytes_per_minute` rather than `bytes_per_meter`.

Also, not all SI units are implemented, because no receiver was using them (like Candela, or Mole). Non-SI units were also added because they were likely to be used, like `Cel` for `celsius`, or `Hz` for `hertz`.

This list of units could totally be expanded at some point, but you're right that it's a breaking change so we should be cautious with this.

Note: as I understand, this logic has been copy-pasted to Prometheus itself, when ingesting native OTLP metrics (feature will be GA with version 3), so we should probably stick to it.

Reply:
Thanks for the context. You're right, this is [copied into the Prometheus server](https://github.com/prometheus/prometheus/blob/c0221d9739c9aa85f43c1b74590ea0af7590b8a7/storage/remote/otlptranslator/prometheus/normalize_name.go#L68-L78). Given that Prometheus itself supports this behavior, I think we should go ahead with it. Sounds good @ocelotl ?

Reply:
All right, let's go ahead.